### PR TITLE
Remove exception for missing clang 3.8 toolchain; update CI config.

### PR DESCRIPTION
**Pull request by dilijev**
Tested these changes in the linux-ci branch and no breaks resulted.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/982)

### CODE REVIEW

**Comment by dilijev** [comment]
FYI @digitalinfinity @obastemur @jianchun 

See #969 #966

File: netci.groovy [hunk 1]
```diff
@@ -122,6 +122,8 @@ def CreateLinuxBuildTasks = { machine, configTag, linuxBranch, nonDefaultTaskSet
             // params: Project, BaseTaskName, IsPullRequest (appends '_prtest')
             def jobName = Utilities.getFullJobName(project, config, isPR)
 
+            def testableConfig = buildType in ['debug']
+
             def infoScript = 'bash jenkins/get_system_info.sh'
             def debugFlag = buildType == 'debug' ? '--debug' : ''
```
Comment (obastemur) [inline netci.groovy]: Do you think that the `buildType`, `debugFlag` and the `testableConfig` above could be arranged/defined a bit more efficiently?

**Comment by obastemur** [comment]
LGTM :+1:

File: netci.groovy (hunk 1 above)
Comment (dilijev) [inline netci.groovy]: I followed the pattern used when there were more options. This works, but I'm open to suggestions.

Comment (dilijev) [inline netci.groovy]: Specifically, @digitalinfinity indicated that there are plans to add a `test` config like we have on Windows, where we will change line 125 to be: `def testableConfig = buildType in ['debug', 'test']`

**Comment by digitalinfinity** [comment]
:shipit:

**Pull request merged**